python/ray/_private/ray_option_utils.py

```diff
@@ -261,6 +261,18 @@ def validate_actor_options(options: Dict[str, Any], in_options: bool):
 
     if options.get("get_if_exists") and not options.get("name"):
         raise ValueError("The actor name must be specified to use `get_if_exists`.")
+
+    if "object_store_memory" in options:
+        warnings.warn(
+            "Setting 'object_store_memory'"
+            " for actors is deprecated since it doesn't actually"
+            " reserve the required object store memory."
+            f" Use object spilling (https://docs.ray.io/en/{get_ray_doc_version()}/ray-core/objects/object-spilling.html) "  # noqa: E501
+            "instead to bypass the object store memory size limitation.",
+            DeprecationWarning,
+            stacklevel=1,
```

Comment:
This seems not able to surface the warning up to the public API.

Reply:
Yea, it will print the current location instead of the caller. It's a bit tricky to get the stacklevel right for all callers and it can be easily broken during code refactor. I'll leave it as 1 for now and the warning message should be clear enough to let the user know what is deprecated.